index.js

```diff
@@ -47,21 +62,41 @@ const getRealType = (value) => {
     //     getRealType(NaN)        // 'NaN'
     // Use typeof, instanceof and some magic. It's enough to have
     // 12-13 unique types but you can find out in JS even more :)
+    if (Number.isNaN(value)) return 'NaN';
+    if (typeof value === 'number' && !isFinite(value)) return 'Infinity';
+    else
+        return {}.toString
+            .call(value)
+            .match(/\s([a-zA-Z]+)/)[1]
+            .toLowerCase();
 };
 
 const getRealTypesOfItems = (arr) => {
     // Return array with real types of items of given array
+    return arr.map((i) => getRealType(i));
```

Comment:
Можно даже проще
```suggestion
    return arr.map(getRealType);
```
Но в него тогда приедут ещё параметры, так что у тебя более безопасный способ.

И я бы назвал переменную нормально, `i` это каунтер для `for` обычно, что она тут значит?
С осмысленным именем понятнее, что ты тут ожидаешь и передаёшь в `getRealType`, когда все переменные понятно названы, проще найти ошибку. Например, ты видишь где-то `resultMap.forEach(...)`, и сразу понимаешь, что тут косяк, потому что вряд ли массив назвали бы `resultMap `, там наверняка объект ключ-значение.

Reply:
согласен, fixed